underpants.go

```diff
@@ -111,6 +127,7 @@ func (u *user) encode(key []byte) (string, error) {
   if err := json.NewEncoder(w).Encode(u); err != nil {
     return "", err
   }
+  w.Close()
```

Comment:
Gotcha, I'm just worried that closing the encoder (and flushing the buffer) _after_ the string has been turned into a fully formed cookie will mean the cookie is still truncated.

Applying the following patch:

```diff
diff --git a/underpants.go b/underpants.go
index a53950f..c8b2635 100644
--- a/underpants.go
+++ b/underpants.go
@@ -123,10 +123,10 @@ func (u *user) encode(key []byte) (string, error) {
   h := hmac.New(sha256.New, key)
   w := base64.NewEncoder(base64.URLEncoding,
     io.MultiWriter(h, &b))
+  defer w.Close()
   if err := json.NewEncoder(w).Encode(u); err != nil {
     return "", err
   }
-  w.Close()
 
   return fmt.Sprintf("%s,%s",
     base64.URLEncoding.EncodeToString(h.Sum(nil)),
@@ -504,6 +504,11 @@ func setup(c *conf, port int) (*http.ServeMux, error) {
       panic(err)
     }
 
+    _, err = decodeUser(v, key)
+    if err != nil {
+      panic(err)
+    }
+
     http.SetCookie(w, &http.Cookie{
       Name:     userCookieKey,
       Value:    url.QueryEscape(v),


```

then running through the auth flow gave me the following error:

```
2015/07/01 18:05:57 http: panic serving 192.168.122.1:57228: unexpected EOF
```

which is what I was seeing without the `w.Close()` at all.